components/planner/history/StudentHistoryView/CourseAudit.tsx

```diff
@@ -0,0 +1,686 @@
+                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
+                </div>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>
+        ))}
+      </TableRow>
+    </TableHead>
+  );
+}
+
+//stylesheet
+const useToolbarStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      paddingLeft: theme.spacing(2),
+      paddingRight: theme.spacing(1),
+    },
+    highlight:
+      theme.palette.type === 'light'
+        ? {
+            color: theme.palette.secondary.main,
+            backgroundColor: lighten(theme.palette.secondary.light, 0.85),
+          }
+        : {
+            color: theme.palette.text.primary,
+            backgroundColor: theme.palette.secondary.dark,
+          },
+    title: {
+      flex: '1 1 100%',
+    },
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 120,
+    },
+    selectEmpty: {
+      marginTop: theme.spacing(2),
+    },
+    wrapper: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      flexWrap: 'wrap',
+      width: '100%',
+    },
+    wider: {
+      width: '10%',
+    },
+    course: {
+      paddingTop: '22px',
+      width: '20%',
+    },
+  }),
+);
+
+/**
+ * Allows adding and deleting courses
+ * numSelected: the number of rows selected
+ * data: the array of CourseAttempt
+ * selectedState: the list of course names that have been selected
+ * update(CourseAttempt[]): updates the rows
+ * updateNum(string[]): updates the number selected
+ * updateSelectedState(string[]): updates the list of courses selected
+ */
+interface EnhancedTableToolbarProps {
+  numSelected: number;
+  data: CourseAttempt[];
+  selectedState: string[];
+  update(courses: CourseAttempt[]): any;
+  updateNum(courses: string[]): any;
+  updateSelectedState(courses: string[]): any;
+}
+
+/**
+ * Renders a list of MenuItem options for the user to select in the dropdowns
+ * @param array - an array of any type where the indices are rendered as separate options
+ * returns the rendered list of MenuItems
+ */
+function returnOptions(array) {
+  const options = [];
+  for (let i = 0; i < array.length; i++) {
+    options.push(<MenuItem value={array[i]}>{array[i]}</MenuItem>);
+  }
+  return options;
+}
+
+/**
+ * Generates the year up to the next 5 years as an array of integers
+ * returns an integer array
+ */
+function returnYears() {
+  const currentYear = new Date().getFullYear() + 5;
+  const max = currentYear - 1970;
+  const years = Array.from(Array(max).keys());
+  for (let i = 0; i < years.length; i++) {
+    years[i] = years[i] + 1970;
+  }
+  return years.reverse();
+}
+
+//The implemented toolbar
+const EnhancedTableToolbar = ({
+  numSelected,
+  data,
+  update,
+  updateNum,
+  selectedState,
+  updateSelectedState,
+}: EnhancedTableToolbarProps) => {
+  const classes = useToolbarStyles();
+  const [adding, setAdding] = React.useState(false);
+  const [courseName, setName] = React.useState('PLACEHOLDER');
```

Comment:
Don't use a placeholder. Just use an empty string:
```suggestion
  const [courseName, setName] = React.useState('');
```